src/components/screens/admin-vacation-management/EditVacationDialog.tsx

```diff
@@ -0,0 +1,79 @@
+import type React from 'react';
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  Button,
+  TextField,
+  Typography,
+  Box,
+  CircularProgress
+} from '@mui/material';
+import type { VacationDays, User } from '../../../types/index';
+
+interface EditVacationDialogProps {
+  open: boolean;
+  user: User | null;
+  vacationDays: VacationDays;
+  loading: boolean;
+  onClose: () => void;
+  onChange: (year: string, field: 'total' | 'remaining', value: string) => void;
+  onSave: () => void;
+  disableSave: boolean; 
+}
+
+const EditVacationDialog: React.FC<EditVacationDialogProps> = ({
```

Comment:
TSDoc comments and type the props in the same way as in other components e.g vacation-requests-table-toolbar.tsx